docs/user_guide/upgrading.rst

```diff
@@ -0,0 +1,38 @@
+Upgrading
```

Comment:
I love this! We should have done this from the start...
Clearly, we need to add a few more sections, perhaps for every new version. I don't expect that to be part of this PR, though. I can sift through the CHANGES.md afterwards and find the API changes that happened between versions.

Reply:
I don't think it's worth spending any time on this now, especially retroactively.  I'm thinking of this as a chore done in the release process.